src/popup/components/TransactionList.vue

```diff
@@ -73,133 +63,150 @@ export default {
+                || (isFungibleTokenTx(tr)
+                  && compareCaseInsensitive(tr.tx.type, SCHEMA.TX_TYPE.contractCall)
+                  && tr.recipient === account.value.address);
+            case 'tips':
+              return (tr.tx.contractId
+                && (activeNetwork.value.tipContractV1 === tr.tx.contractId
+                  || activeNetwork.value.tipContractV2 === tr.tx.contractId)
+                && (tr.tx.function === 'tip' || tr.tx.function === 'retip')) || tr.claim;
+            default:
+              throw new Error(`${root.$t('pages.recentTransactions.unknownMode')} ${displayMode.value.filter}`);
+          }
+        })
+        .filter(
+          (tr) => !searchBar.value
+            || getTxSymbol.value(tr)
+              .toLocaleLowerCase()
+              .includes(searchBar.value.toLocaleLowerCase()),
+        )
+        .sort((a, b) => {
+          const arr: any[] = [a, b].map((tr) => new Date(tr.microTime));
+          if (displayMode.value.rotated) arr.reverse();
+
+          return (arr[0] - arr[1]) || a.pending;
+        })
+        .slice(0, props.maxLength || Infinity);
     });
-    this.$watch(({ displayMode }) => displayMode, this.checkLoadMore);
-  },
-  methods: {
-    checkLoadMore() {
-      if (this.isDestroyed || !this.transactions.nextPageUrl) return;
-      const isDesktop = document.documentElement.clientWidth > 480 || process.env.IS_EXTENSION;
-      const { scrollHeight, scrollTop, clientHeight } = isDesktop
-        ? document.querySelector('.app-inner') : document.documentElement;
-      if (this.maxLength && this.filteredTransactions.length >= this.maxLength) return;
-      if (scrollHeight - scrollTop <= clientHeight + 100) this.loadMore();
-    },
-    compareCaseInsensitive(str1, str2) {
-      return str1.toLocaleLowerCase() === str2.toLocaleLowerCase();
-    },
-    async loadMore() {
-      if (this.loading) return;
-      this.loading = true;
+
+    const showSearchAndFilters = computed(() => props.showFilters
+      || displayMode.value.filter !== 'all'
+      || searchBar.value);
+
+    const loadMore = async () => {
+      if (loading.value) return;
+      loading.value = true;
       try {
-        await watchUntilTruthy(() => this.$store.state.middleware);
-        await this.$store.dispatch('fetchTransactions', { limit: TXS_PER_PAGE });
+        await watchUntilTruthy(() => root.$store.state.middleware);
+        await root.$store.dispatch('fetchTransactions', { limit: TXS_PER_PAGE });
       } finally {
-        this.loading = false;
+        loading.value = false;
       }
-      this.checkLoadMore();
-    },
-    async getLatest() {
+      // eslint-disable-next-line no-use-before-define
+      checkLoadMore();
+    };
+
+    const checkLoadMore = () => {
+      if (isDestroyed.value || !transactions.value.nextPageUrl) return;
+      const isDesktop = (document.documentElement.clientWidth > 480 || process.env.IS_EXTENSION);
```

Comment:
You could use `viewport` composable from #1730. It will allow you to remove the whole logic related to scrolling and listening for it